spec/security/modprobe_spec.rb

```diff
@@ -0,0 +1,79 @@
+describe LinuxAdmin::Security::Modprobe do
+  def test_file_name
+    File.join(data_file_path("security"), "modprobe_lockdown")
+  end
+
+  def new_file_name
+    File.join(data_file_path("security"), "modprobe_new")
+  end
+
+  def test_file_contents
+    File.read(test_file_name)
+  end
+
+  def new_file_contents
+    File.read(new_file_name)
+  end
+
+  around(:each) do |example|
+    text = test_file_contents
+    example.run
+    File.write(test_file_name, text)
+  end
+
+  describe "#apply_scap_settings" do
+    it "disables the dccp protocol" do
+      described_class.new.apply_scap_settings(test_file_name)
+      expect(test_file_contents).to match(%r{^install dccp /bin/true\n})
+    end
+
+    it "disables the sctp protocol" do
+      described_class.new.apply_scap_settings(test_file_name)
+      expect(test_file_contents).to match(%r{^install sctp /bin/true\n})
+    end
+
+    it "disables the rds protocol" do
+      described_class.new.apply_scap_settings(test_file_name)
+      expect(test_file_contents).to match(%r{^install rds /bin/true\n})
+    end
+
+    it "disables the tipc protocol" do
+      described_class.new.apply_scap_settings(test_file_name)
+      expect(test_file_contents).to match(%r{^install tipc /bin/true\n})
+    end
+  end
+
+  describe "#disable_module" do
+    it "adds a module to the file" do
+      described_class.new.disable_module("test_module", test_file_name)
+      expect(test_file_contents).to match(%r{^install test_module /bin/true\n})
+    end
+
+    context "creates a file" do
+      after do
+        File.exist?(new_file_name) && File.delete(new_file_name)
+      end
+      it "if the given one doesn't exist" do
+        expect(File.exist?(new_file_name)).to be false
+        described_class.new.disable_module("test_module", new_file_name)
+        expect(File.exist?(new_file_name)).to be true
```

Comment:
Technically, yes, but with this line we would get a failed test rather than an exception if the file doesn't get created.